harvdev_utils/chado_functions/synonym.py

```diff
@@ -60,15 +60,15 @@ def synonym_name_details(session: Session, synonym_name: str, nosup: bool=False)
         try:
             organism = get_organism(session, short=abbr)
         except CodingError:  # Not a species abbr so continue as normal
-            return get_default_organism(session), sgml_to_plain_text(synonym_name), sgml_to_unicode(sub_sup_to_sgml(synonym_name))
+            return get_default_organism(session), sgml_to_plain_text(greek_to_sgml(synonym_name)), sgml_to_unicode(sub_sup_to_sgml(synonym_name))
```

Comment:
OK, I think I can see how the problem arose now, and how this merge would fix it.
I do have a question. It seems that when this `synonym_name_details` function is used, the input `synonym_name` is full unicode (so would have Greek chars in it). I guess I'm surprised by that given that curators do not submit Greek chars, but maybe there's an earlier step in the proforma parser that converts all SGML (e.g., `&agr;`) to proper unicode char? Is that how it works?

Reply:
Except for the chemicals the synonyms always come from the curators as &agr; but for chemicals we get synonyms form the chebi and pubchem too which have greek chars in.

Reply:
A lot of tests are failing now that use this so something is not correct. Probably be an update later.